src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -214,8 +252,16 @@ def __format__(self, __format_spec: str) -> str:
         if math.isinf(self._base_value) or math.isnan(self._base_value):
             return f"{self._base_value} {self._exponent_unit_map[0]}"
 
-        abs_value = abs(self._base_value)
-        exponent = math.floor(math.log10(abs_value)) if abs_value else 0
+        if abs_value := abs(self._base_value):
+            precision_pow = 10 ** (precision)
+            if math.isclose(abs_value, precision_pow, abs_tol=1e-4, rel_tol=0.01):
+                # If the value is close to the precision, round it
+                exponent = math.ceil(math.log10(precision_pow))
+            else:
+                exponent = math.floor(math.log10(abs_value))
+        else:
+            exponent = 0
+
```

Comment:
I think it would be good to explain in the commit message why is this needed, I was surprised to see changes in `format()` when the commit says it only adds `from_string()`. Maybe even split the commit if appropriate (I see also there are new tests for `format()`, so it seems like a bug fix that sneaked in.

Thinking about it might be good too to include a small comment in the code about this.